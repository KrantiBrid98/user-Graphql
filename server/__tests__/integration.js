const { createTestClient } = require('apollo-server-testing');
const gql = require('graphql-tag');
const { ApolloServer } = require('apollo-server');


// const { constructTestServer } = require('../__utils');
const {
    typeDefs,
    resolvers,
} = require('../server')


const USER_QUERY = gql`   
{
    users{
      firstName
      id
    }
}
`

const ADD_USER = gql`
mutation addUser($firstName:String!) {
    addUser(firstName: $firstName){
        id
        firstName
    }
  }
`;

const server = new ApolloServer({
    typeDefs,
    resolvers
});

describe('Query', () => {
    it('fetch list of users', async () => {
        const { query } = createTestClient(server);
        const res = await query({ query: USER_QUERY });
        expect(res).toMatchSnapshot();
    });
})

describe('Mutation', () => {
    it('add user to list', async () => {
        const { mutate } = createTestClient(server)
        const res = await mutate({
            mutation: ADD_USER,
            variables: { firstName: 'testUser' }
        });
        expect(res).toMatchSnapshot();
    });
})