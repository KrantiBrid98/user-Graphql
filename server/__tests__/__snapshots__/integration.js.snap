// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mutation add user to list 1`] = `
Object {
  "data": Object {
    "addUser": null,
  },
  "errors": Array [
    [GraphQLError: Network Error],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Query fetch list of users 1`] = `
Object {
  "data": Object {
    "users": null,
  },
  "errors": Array [
    [GraphQLError: Network Error],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
